### ISSUE TITLE
brew info --analytics does not handle extra args

### ISSUE THREAD

**Issue by fxcoudert** [Homebrew/brew#6677]
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew config` and `brew doctor` and included their output with your issue?

## Bug description

```
$ brew info --analytics --category=install --days=90 mpfr
==> Analytics
==> install (30 days)
Index | Name (with options)                                                                                      |  Count |  Percent
-----:|----------------------------------------------------------------------------------------------------------|-------:|--------:
1     | mpfr                                                                                                     | 66,978 |  100.00%
==> install (90 days)
Index | Name (with options)                                                                                     |   Count |  Percent
-----:|---------------------------------------------------------------------------------------------------------|--------:|--------:
1     | mpfr                                                                                                    | 170,352 |  100.00%
==> install (365 days)
Index | Name (with options)                                                                                     |   Count |  Percent
-----:|---------------------------------------------------------------------------------------------------------|--------:|--------:
1     | mpfr                                                                                                    | 744,390 |  100.00%
==> install_on_request (30 days)
Index | Name (with options)                                                                                       | Count |  Percent
-----:|-----------------------------------------------------------------------------------------------------------|------:|--------:
1     | mpfr                                                                                                      |   754 |  100.00%
==> install_on_request (90 days)
Index | Name (with options)                                                                                       | Count |  Percent
-----:|-----------------------------------------------------------------------------------------------------------|------:|--------:
1     | mpfr                                                                                                      | 2,153 |  100.00%
==> install_on_request (365 days)
Index | Name (with options)                                                                                      |  Count |  Percent
-----:|----------------------------------------------------------------------------------------------------------|-------:|--------:
1     | mpfr                                                                                                     | 14,851 |  100.00%
==> build_error (30 days)
Index | Name (with options)                                                                                       | Count |  Percent
-----:|-----------------------------------------------------------------------------------------------------------|------:|--------:
1     | mpfr                                                                                                      |     0 |    0.00%
```

I expect that, given then `--category` and `--days` arguments, only the relevant information will be printed.

**Comment by EricFromCanada** [comment]
What isn't entirely clear from the man page is that there are four modes for the command: 1) `brew info`, 2) `brew info <formula>`, and two modes for `--analytics`: 3) formula-specific analytics (when a formula is specified), and 4) global analytics, for which `--days` and `--category` apply to.

**Comment by EricFromCanada** [comment]
Fixed in #6714.
